### PR TITLE
Homework7

### PR DESCRIPTION
**Pull request by KatoCimoto**

### CODE REVIEW

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                }
+                System.out.println(guessChars);
+            }
+        }
+        System.out.println("Congratulations! You guessed the word: " + answer);
+    }
+
+    //pt 5
+    public static boolean isPalindrome(String str){
+        StringBuilder strBuilder = new StringBuilder(str);
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java]: here you could reuse your stringReverse() method )

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    //pt 4
+    public static String stringReverse(String str){
+        return new StringBuilder(str).reverse().toString();
+    }
+
+    //pt 3
+    public static int findWordPosition(String source, String target){
+        int position = source.indexOf(target);
+        if (position == -1) {
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java]: no reasons in this validation, you could simply `return source.indexOf(target)`

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        }
+    }
+
+
+    // pt 2
+    public static void findSymbolOccurance(String str, char ch){
+
+        int count = 0;
+        char[] strArray = str.toCharArray();
+        for (char c : strArray) {
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java]: not a mistake, but IMO simply `for (char c : strArray.toCharArray())`  looks better

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            if (ch == c) {
+                count++;
+            }
+        }
+        System.out.println("The number of char \"" + ch + "\" in a string is " + count);
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java]: also not a mistake, but you can use  single quotation marks  ' ', it becomes more readable

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                "pumpkin", "potato"};
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Hello! You need to guess one word that I thought of.");
+        System.out.println("This word refers to the theme of vegetables and fruits.");
+
+        Random random = new Random();
+
+        String answer = words[random.nextInt(words.length)];
+        char[] answerChars = answer.toCharArray();
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java]: dont create separate char arrays

**Review by agrok — CHANGES_REQUESTED** [review]

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java [hunk 6]
```diff
@@ -8,50 +8,53 @@ public class Main {
...
-            String guess = scanner.nextLine().toLowerCase();
+            System.out.print("Enter your guess (lowercase only): ");
+            String answer = scanner.nextLine();
 
-            if (guess.equals(answer)) {
-                guessed = true;
+            if (answer.equals(word)) {
+                System.out.println("Congratulations! You guessed the word in " + attempts + " attempts.");
+                break;
             } else {
```
Comment (agrok) [inline src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java:38]: if you have break in `if`, you dont need `else` here

**Review by agrok — APPROVED** [review]

File: src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java (hunk 6 above)
Comment (KatoCimoto) [reply src/main/java/com/hillel/shabrin/homeworks/homework7/Main.java:38]: yeah, I know that. just gotta get used to it. thanks :)

**Pull request closed without merging**